### PR TITLE
Add Coderabbit config

### PR DESCRIPTION
**Pull request by dagardner-nv**
## Description
* Add an initial `.coderabbit.yaml` 
* The YAML file primarily consists of two types of items:
  - instructions: Prompts for the LLM on how to review PRs
  - Enabling/disabling specific features

## By Submitting this PR I confirm:
- I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/NeMo-Agent-Toolkit/blob/develop/docs/source/resources/contributing.md).
- We require that all contributors "sign-off" on their commits. This certifies that the contribution is your original work, or you have rights to submit it under the same license, or a compatible license.
  - Any contribution which contains commits that are not Signed-Off will not be accepted.
- When the PR is ready for review, new or existing tests cover these changes.
- When the PR is ready for review, the documentation is up to date with these changes.

## Summary by CodeRabbit

* **Chores**
  * Added repository-wide automated code-review configuration with license header and language setting.
  * Configured pre-merge checks, enabled auto-review (incremental, drafts disabled) and auto-labeling with defined label rules.
  * Integrated linters and scanners (ruff, yamllint, markdownlint, gitleaks, osvScanner).
  * Expanded path-based guidance for docs, examples, core, packages, and tests.
  * Enabled sequence diagrams; added top-level poem flag and disabled suggested reviewers.

### CODE REVIEW

File: .coderabbit.yaml [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+      instructions: >-
+        Apply this label when the PR contains invalid changes, doesn't follow project guidelines, or has fundamental
+        issues that make it unsuitable for the project.
+    - label: non-breaking
+      instructions: >-
+        Apply this label when the PR introduces changes that maintain backward compatibility, does not create a
+        behavioral change, and does not require users to modify their existing code. If you apply this label, do not
+        apply the "breaking" label.
+
+  sequence_diagrams: false
```
Comment (willkill07) [inline .coderabbit.yaml]: I like the sequence diagrams :)

File: .coderabbit.yaml [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+      mode: warning
+      requirements: "Title should be concise, and descriptive, and use imperative mood (max ~72 chars)."
+    description:
+      mode: warning
+    issue_assessment:
+      mode: warning
+  auto_review:
+    enabled: true
+    drafts: false
+    auto_incremental_review: false
```
Comment (willkill07) [inline .coderabbit.yaml]: I know we can force an incremental review via `@coderabbitai review` but it would be nice to have that documented somewhere

**Comment by dagardner-nv** [comment]
> FYI, Dynamo's config is here: https://github.com/ai-dynamo/dynamo/blob/main/.coderabbit.yaml
> 
> Looks like we can also turn off poem

That's what I started with, they left the poem on :)

File: .coderabbit.yaml [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+          `tests/` directory at the same level as the `pyproject.toml` file.
+
+    - path: "tests/**/*.py"
+      instructions: >-
+        - Ensure that tests are comprehensive, cover edge cases, and validate the functionality of the code.
+        - Test functions should be named using the `test_` prefix, using snake_case.
+        - Any frequently repeated code should be extracted into pytest fixtures.
+        - Pytest fixtures should define the name argument when applying the pytest.fixture decorator. The fixture
+          function being decorated should be named using the `fixture_` prefix, using snake_case. Example:
+            @fixture(name="my_fixture")
```
Comment (willkill07) [inline .coderabbit.yaml]:
```suggestion
            @pytest.fixture(name="my_fixture")
```

File: .coderabbit.yaml [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+      instructions: >-
+        Apply this label when the PR primarily adds, updates, or improves documentation including README files. If you
+        apply this label do not apply the "bug", "feature request", or "improvement" labels.
+    - label: duplicate
+      instructions: >-
+        Apply this label when the PR addresses the same issue or implements the same feature as another existing
+        PR or issue.
+    - label: external
+      instructions: >-
+        Apply this label when the PR was submitted by someone who is not a member of the @NVIDIA/agentiq-developers
```
Comment (willkill07) [inline .coderabbit.yaml]: The coderabbit suggestion from its review was pretty good:
> GitHub exposes author_association (OWNER, MEMBER, COLLABORATOR, CONTRIBUTOR, FIRST_TIMER, NONE). Suggest basing guidance on that.
```
Apply this label when the PR author is not an organization OWNER or MEMBER (i.e., author_association is COLLABORATOR, CONTRIBUTOR, FIRST_TIMER, FIRST_TIME_CONTRIBUTOR, or NONE).
```

**Review by willkill07 — APPROVED** [review]
Approving with nits
